sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerClientBuilder.java

```diff
@@ -0,0 +1,354 @@
+            policies.add(new AddHeadersPolicy(headers));
+            policies.add(new AddDatePolicy());
+
+            HttpPolicyProviders.addBeforeRetryPolicies(policies);
+            policies.add(retryPolicy == null ? DEFAULT_RETRY_POLICY : retryPolicy);
+            policies.addAll(this.policies);
+            HttpPolicyProviders.addAfterRetryPolicies(policies);
+
+            policies.add(new HttpLoggingPolicy(httpLogOptions));
+
+            pipeline = new HttpPipelineBuilder()
+                .policies(policies.toArray(new HttpPipelinePolicy[0]))
+                .httpClient(httpClient)
+                .build();
+        }
+
+        final FormRecognizerClientImpl formRecognizerAPI = new FormRecognizerClientImplBuilder()
+            .endpoint(endpoint)
+            .pipeline(pipeline)
+            .build();
+
+        return new FormRecognizerAsyncClient(formRecognizerAPI, serviceVersion);
+    }
+
+    /**
+     * Sets the service endpoint for the Azure Form Recognizer instance.
+     *
+     * @param endpoint The URL of the Azure Form Recognizer instance service requests to and receive responses from.
+     * @return The updated FormRecognizerClientImplBuilder object.
+     * @throws NullPointerException if {@code endpoint} is null
+     * @throws IllegalArgumentException if {@code endpoint} cannot be parsed into a valid URL.
+     */
+    public FormRecognizerClientBuilder endpoint(String endpoint) {
+        Objects.requireNonNull(endpoint, "'endpoint' cannot be null.");
+
+        try {
+            new URL(endpoint);
+        } catch (MalformedURLException ex) {
+            throw logger.logExceptionAsWarning(new IllegalArgumentException("'endpoint' must be a valid URL.", ex));
+        }
+
+        if (endpoint.endsWith("/")) {
+            this.endpoint = endpoint.substring(0, endpoint.length() - 1);
+        } else {
+            this.endpoint = endpoint;
+        }
+
+        return this;
+    }
+
+    /**
+     * Sets the credential to use when authenticating HTTP requests for this FormRecognizerClientImplBuilder.
+     *
+     * @param apiKeyCredential API key credential
+     *
+     * @return The updated FormRecognizerClientImplBuilder object.
+     * @throws NullPointerException If {@code apiKeyCredential} is {@code null}
+     */
+    public FormRecognizerClientBuilder apiKey(FormRecognizerApiKeyCredential apiKeyCredential) {
+        this.credential = Objects.requireNonNull(apiKeyCredential, "'apiKeyCredential' cannot be null.");
+        return this;
+    }
+
+    /**
+     * Sets the {@link TokenCredential} used to authenticate HTTP requests.
+     *
+     * @param tokenCredential TokenCredential used to authenticate HTTP requests.
+     * @return The updated FormRecognizerClientImplBuilder object.
+     * @throws NullPointerException If {@code tokenCredential} is {@code null}.
+     */
+    public FormRecognizerClientBuilder credential(TokenCredential tokenCredential) {
```

Comment:
Does FormRecognizer not support `connectionString`?

Reply:
Yes, similar to Text Analytics this doesn't have connection string authentication.
Also, currently, Form recognizer does not support AAD.